src/sympc/tensor/mpc_tensor.py

```diff
@@ -750,8 +763,12 @@ def _get_shape(
         else:
             op = getattr(operator, op_str)
 
-        x = torch.empty(size=x_shape)
-        y = torch.empty(size=y_shape)
+        if op_str == "xor":
+            x = torch.empty(size=x_shape, dtype=torch.bool)
```

Comment:
Q: Why do you need this?
We do this operation only to check the shape

Reply:
torch.empty generates float by default for xor we need bool.

Reply:
Q: Then why don't we keep only the `bool` case?

Reply:
Since we do rshift operation, in ABY3 truncation, shift does not work for torch.bool george.